### PR TITLE
WIP:fix: bringing spectral package back to passing KFM

### PR DESCRIPTION
**Pull request by dimakis**
DO NOT MERGE.

### CODE REVIEW

**Comment by dimakis** [comment]
@wtrocki I took a chainsaw to this. With these changes spectral lint will pass KFM with no errors or warnings. 
note that to do so I had to delete 4 specific RHOAS rules as outlined in the comments in the ruleset. 
I also had to turn off 6 more oas rules. 4 for warnings and 2 for errors.
 
obviously this is not the solution we are looking for. do you want to move ahead with these or should I keep the oas rules?

File: spectral/ruleset.yaml [hunk 1]
```diff
@@ -9,6 +9,21 @@ functions:
 rules:
   openapi-tags: off
   operation-tags: off
+  # openapi spec rules, that return errors
+  oas3-valid-schema-example: off
```
Comment (wtrocki) [inline spectral/ruleset.yaml:13]: -1 to disable it. That should be trivial fix on any API we have.

Comment (dimakis) [reply spectral/ruleset.yaml:13]: ok so keep those in there and fix up the KFM etc?

File: spectral/ruleset.yaml [hunk 2]
```diff
@@ -9,6 +9,21 @@ functions:
...
+  no-$ref-siblings: off
```
Comment (wtrocki) [inline spectral/ruleset.yaml:14]: I think that one would not be required - we need to update to openapi 3.1 where this is supported. Linter should probably stop reporting it.

File: spectral/ruleset.yaml [hunk 3]
```diff
@@ -9,6 +9,21 @@ functions:
...
+  # openapi spec rules, that return warnings
+  operation-description: off
```
Comment (wtrocki) [inline spectral/ruleset.yaml:16]: This is trivial fix that we should enforce.

File: spectral/ruleset.yaml [hunk 4]
```diff
@@ -9,6 +9,21 @@ functions:
...
+  operation-tag-defined: off
+  info-contact: off
```
Comment (wtrocki) [inline spectral/ruleset.yaml:18]: Why. This is trivial to have

File: spectral/ruleset.yaml (hunk 3 above)
Comment (wtrocki) [reply spectral/ruleset.yaml:16]: yes.. But that is wrong way to fix it. Ruleset should stay as it is. If we do not want to fix anything we should add exceptions per service in their spectral.yaml as suggested to RHOSE etc.

Comment (dimakis) [reply spectral/ruleset.yaml:16]: ok I got you now. I thought you wanted our ruleset changed to pass. thats what I got from the chat. no worries. I can sort this.

**Pull request closed without merging**